### PR TITLE
Adding merge string alternatively in C-Plus-Plus/cp

### LINKED ISSUE #4104: Merge Strings Alternatively [C-Plus-Plus/cp]
## 💥 Proposal

Given two strings word1 and word2. Merge the strings by adding letters in alternating order, starting with word1. If a string is longer than the other, append the additional letters onto the end of the merged string.
Return the merged string.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?
Yes

### PR DESCRIPTION
**Pull request by AyushPradhan9**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
Yes

### Description
Adding merge string alternatively algorithm in C-Plus-Plus/cp

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
Closes: #4104

### CODE REVIEW

File: C-Plus-Plus/cp/Merge_string_alternatively.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+Explanation:
+First we alternatively add both the strings to the resultant string till the length of the smallest string (both strings can also be equal length). 
+After that we add the substring of the remaining longer string to resultant string if the length of both strings are not equal.
+**/
+
+#include<bits/stdc++.h>
+using namespace std;
+
+string mergeAlternately(string word1, string word2) {
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Merge_string_alternatively.cpp:25]: Add one line function description

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]
Few changes: 

- Add sample I/O at the end of the Code
- Add Time and Space Complexity at the end of the Code   
- Add new line at the EOF

**Comment by AyushPradhan9** [comment]
Hey, @ankitaggarwal23 I have pushed the requested changes. Regards!

File: C-Plus-Plus/cp/Merge_string_alternatively.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+After that we add the substring of the remaining longer string to resultant string if the length of both strings are not equal.
+
+**/
+
+#include<bits/stdc++.h>
+using namespace std;
+
+string mergeAlternately(string word1, string word2) {
+
+    /* This function will take 2 string as input and return the resultant merged string formed alternatively from the inputted strings */
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Merge_string_alternatively.cpp]: Shift the comment above the function

File: C-Plus-Plus/cp/Merge_string_alternatively.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    string ans="";
+    int less=min(word1.length(),word2.length());    // takes the size of the smallest string among word1 and word2
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Merge_string_alternatively.cpp]: Shift the comment above the line of code

File: C-Plus-Plus/cp/Merge_string_alternatively.cpp [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    int i=0,j=0; 
+
+    while(i<less || j<less){    // alternatively add word by word from each string to resultant string
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Merge_string_alternatively.cpp]: Shift the comment above the line of code

File: C-Plus-Plus/cp/Merge_string_alternatively.cpp [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+            ans+=word1[i];
+            i++;
+        }
+        else{
+            ans+=word2[j];
+            j++;
+        }
+    }
+
+    if(word1.length()!=word2.length()){     // if both strings are of unequal sizes then add the rest of substring to the resultant string
```
Comment (ankitaggarwal23) [inline C-Plus-Plus/cp/Merge_string_alternatively.cpp]: Shift the comment above the line of code

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

**Comment by AyushPradhan9** [comment]
Thanks for the guidance!

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**